mlir/tools/mlir-tblgen/OpDefinitionsGen.cpp

```diff
@@ -3088,6 +3088,26 @@ void OpEmitter::buildParamList(SmallVectorImpl<MethodParameter> &paramList,
       defaultValuedAttrStartIndex = i;
     }
   }
+
+  // Check if parameters besides default valued one are enough to distinguish
+  // between builders with wrapped and unwrapped arguments.
+  bool hasBuilderAmbiguity = true;
+  for (int i = 0; i < op.getNumArgs(); ++i) {
```

Comment:
nit: don't recompute the end after every iteration: https://llvm.org/docs/CodingStandards.html#don-t-evaluate-end-every-time-through-a-loop

You could probably also turn this into a range for loop